### PR TITLE
Add Loki smoketests to CI

### PR DESCRIPTION
**Pull request by vyzigold**
This PR adds smoketests for Loki testing to the CI. There are changes to the Jenkinsfile to use the loki-operator to deploy minimal flavor of Loki (only the operator was deployed before). And then a similar approach as logging integration testing in sg-core is used - I used an rsyslog container to generate some logs, which are then checked, if they arrived to Loki.

### CODE REVIEW

**Comment by vyzigold** [comment]
So far I tested this only locally in crc, where it worked, I still need to test it here and I expect some issues to pop up, so please don't spend too much time on reviewing right now. I also see I need to do a rebase.

**Comment by vyzigold** [comment]
test

**Comment by vyzigold** [comment]
I rebased on top of master and solved all the merge conflicts

**Comment by vyzigold** [comment]
test

**Comment by vyzigold** [comment]
test

**Comment by vyzigold** [comment]
Last time PSI had trouble cloning the repo, let's try this again. I also rebased on top of current master.

**Comment by vyzigold** [comment]
test

**Comment by vyzigold** [comment]
test

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
test

**Comment by leifmadsen** [comment]
test

**Comment by elfiesmelfie** [comment]
test

File: tests/smoketest/smoketest_job.yaml.template [hunk 1]
```diff
@@ -5,7 +5,6 @@ metadata:
 spec:
   parallelism: 1
   completions: 1
-spec:
```
[the remark this answers is not in the document]
Comment (csibbitt) [reply tests/smoketest/smoketest_job.yaml.template:8]: I think it's probably just redundant; the keys in this spec block would  be merged with the keys from the spec block started a couple lines above on L5

**Comment by vyzigold** [comment]
The CI finally ran, I think I actually encountered the problem, that's currently there. This should solve the issue when it's merged: https://github.com/infrawatch/smart-gateway-operator/pull/106

File: tests/smoketest/smoketest_job.yaml.template (hunk 1 above)
[the remark this answers is not in the document]
Comment (vyzigold) [reply tests/smoketest/smoketest_job.yaml.template:8]: I saw this and thought the line is redundant, so I tried to remove it and everything worked the same, so I left it like this. Now that you pointed this out, I agree this is unrelated to this PR and I should have either leave it be or put it into another PR.

**Comment by andyvesel** [comment]
test

**Comment by csibbitt** [comment]
This is pretty old now but I'm cleaning up the PR queue. Ultimately, I don't think we are ever going to support loki as part of STF itself, but this work is still potentially valuable in rsyslog testing. 

I really don't want any loki related issues ever blocking our CI, so I'm putting the do-not-merge label on this for now.

What I think would be most useful is if we had a nice clean flag to enable/disable loki in all builds, tests, etc, and then do a pass over all the code again to make sure we've encapsulated the loki related pieces as much as possible and are not littering the code with conditions to turn it off in 1000 places.

I'm going to send this to jenkins just to check the status, and because I've been exercising and observing the CI today

**Review by csibbitt — CHANGES_REQUESTED** [review]
This is all good work that I'd like to keep around, but it needs to be cleanly disabled by default

**Comment by csibbitt** [comment]
test

**Comment by vyzigold** [comment]
I agree, that the pull request doesn't make sense in it's current form anymore. If we want to keep this around, maybe we could add another trigger for the CI, that would run build and tests together with Loki.

So for example if somebody would write "test loki" here in the comments, a new job (or however it's called in jenkins) would trigger, similar to what runs when you write "test" right now, but it would run the stf-run-ci with "__deploy_loki_enabled" and other loki specific variables. It would also pass an argument to build/validate_deployment.sh and tests/smoketest/smoketest.sh and those scripts would execute the Loki related lines from this PR only if that argument is present.

Another idea is, because it's really unlikely to ever continue in the Loki efforts in STO, I could just go through this PR, remove the parts, that try to deploy or test Loki and leave only the rsyslog related changes (probably just the smoketest_rsyslog_entrypoint.sh and rsyslog.conf files), which for now would just stay in the repository without ever being executed, but they would be here if we ever decide to use them.

**Comment by csibbitt** [comment]
>maybe we could add another trigger for the CI, that would run build and tests together with Loki. 
> So for example if somebody would write "test loki"...

Personally I don't think this is a necessary part of STO CI at all and don't think it's worth the effort to define and maintain a second trigger. I am more interested in preserving our ability to run it locally in order to validate rsyslog -> loki which still sounds like it is our future path for logging.

> Another idea is, because it's really unlikely to ever continue in the Loki efforts in STO, I could just go through this PR, remove the parts, that try to deploy or test Loki and leave only the rsyslog related changes (probably just the smoketest_rsyslog_entrypoint.sh and rsyslog.conf files), 

Having some quick and easy way to deploy Loki AND rsyslog and test them together is what I'm interested in preserving. This really doesn't have anything to do with STO, though, so perhaps trying to separate out all of the logging related code into it's own repo (we can brainstorm names in chat?) might be a good approach.

CC: @mrunge @leifmadsen

**Comment by leifmadsen** [comment]
Based on our road map and other work recently, should we just close this out?

**Pull request closed without merging**